### PR TITLE
React state

### PR DESCRIPTION
**Pull request by samlee0258**

### CODE REVIEW

File: react-state/my-app/src/ToggleButton.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  const [isClicked, setIsClicked] = useState(false);
+
+  console.log('useState:', isClicked); // false on mount; true on re-render
+  function handleClick() {
+    console.log('before setter:', isClicked); // false
+    setIsClicked(!isClicked);
+    console.log('after setter:', isClicked); // false because the data referencing the state hasn't updated.
+  }
+  // either returns on line 14 or 22 re-renders the ToggleButton component, re-running the code.
+  if (!isClicked) {
```
Comment (RobertGardner) [inline react-state/my-app/src/ToggleButton.js:13]: This works, but it duplicates a lot of code. See if you can figure out how to avoid the duplication.

**Review by RobertGardner — CHANGES_REQUESTED** [review]
Good, but fix the duplication before I approve.

File: react-state/my-app/src/ToggleButton.js (hunk 1 above)
Comment (samlee0258) [reply react-state/my-app/src/ToggleButton.js]: Screenshot 2023-04-01 at 12 18 06 PM

Comment (samlee0258) [reply react-state/my-app/src/ToggleButton.js]: Removed the duplicated code and combined the styles using a ternary.

**Review by RobertGardner — APPROVED** [review]

**Pull request merged**